substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/TestAllocationRequiringGCEvent.java

```diff
@@ -54,6 +54,9 @@ public void test() throws Throwable {
 
     private static void validateEvents(List<RecordedEvent> events) {
         assertTrue(events.size() > 0);
+        for (RecordedEvent event : events) {
+            checkStackTraceTrimming(event, "emit");
```

Comment:
Shouldn't we trim the "emit" methods as well?

Reply:
I wanted to trim the "emit" methods as well, but when I increment the skipcount, it seems to trim 2 frames. Maybe this is due to inlining? I'm not entirely sure.

Reply:
So I decided it was better to trim slightly less and keep more information in the stacktrace.

Reply:
> it seems to trim 2 frames

Interesting - it seems that the `skipCount` logic is broken at the moment as it is applied on the level of physical stack frames (instead of virtual frames). We need to fix that, otherwise, `skipCount` will be pretty much unpredictable.

We would probably need to do something like the following:
- remove the `skipCount` from `JfrStackWalker` (at that point in time, we only know about the physical stack frames)
- modify `JfrStackWalker` so that it stores the requested skip count at the very beginning of the sampler data (i.e., before any recorded IPs)
- when serializing the raw IPs into JFR stack traces, we need to apply the skip count on the level of Java frames

Please let me know if you can look into that.

Reply:
Yes, I can look into this. I'll add that change to this pull request (unless you think it's cleaner separately).

Reply:
We also probably need to delay when we do truncation to the time of serialization. Otherwise we may truncate unnecessarily.

Reply:
Thanks!

I think that 2 different kinds of truncation is needed for the bottom-most frames:
- `JfrStackWalker`: truncate so that we have at most `skipCount + getMaxDepth()` physical frames (each physical frame corresponds to at least 1 virtual frame)
- during serialization: apply skipCount to remove the top frames and truncate the bottom-most frames so that we have at most `getMaxDepth()` virtual frames.

Reply:
Right, I see. I'll work on adding this. Thanks!